src/coreclr/jit/gentree.cpp

```diff
@@ -30439,7 +30439,7 @@ GenTree* Compiler::gtFoldExprHWIntrinsic(GenTreeHWIntrinsic* tree)
 #if defined(TARGET_XARCH)
         tryHandle = op->OperIsHWIntrinsic();
 #elif defined(TARGET_ARM64)
-        if (op->OperIsHWIntrinsic() && op->OperIsHWIntrinsic(NI_Sve_CreateTrueMaskAll))
+        if (op->OperIsHWIntrinsic() && op->OperIsHWIntrinsic(NI_Sve_CreateTrueMaskAll) && op2->OperIsHWIntrinsic())
```

Comment:
```suggestion
        if (op->OperIsHWIntrinsic(NI_Sve_CreateTrueMaskAll) && op2->OperIsHWIntrinsic())
```

Reply:
I'd probably do it instead as:
```cpp
#elif defined(TARGET_ARM64)
if (op->OperIsHWIntrinsic(NI_Sve_CreateTrueMaskAll))
{
    op        = op2;
    tryHandle = op->OperIsHWIntrinsc();
}
#endif // TARGET_ARM64
```

That makes it generally consistent with how xarch does it and ensures op is the right node